abi/src/main/java/org/web3j/abi/DefaultFunctionEncoder.java

```diff
@@ -64,10 +68,21 @@ private static String encodeParameters(
         return result.toString();
     }
 
+    @SuppressWarnings("unchecked")
     private static int getLength(final List<Type> parameters) {
         int count = 0;
         for (final Type type : parameters) {
-            if (type instanceof StaticArray) {
+            if (type instanceof StaticArray
+                    && StaticStruct.class.isAssignableFrom(
+                            ((StaticArray) type).getComponentType())) {
+                count +=
+                        staticStructCanonicalFieldsCount(((StaticArray) type).getComponentType())
```

Comment:
Is a good idea to make utility methods for general use. 

In this case having a count is very specific, the method `staticStructCanonicalFieldsCount` should return the list of fields. 

Also the name is not clear, the only difference with `staticStructsNestedFieldsFlatList` is that the fields are public.  

So maybe call it something like `staticStructsNestedPublicFieldsFlatList`? To make clear that is the same method - just a suggestion.